api/api.go

```diff
@@ -696,7 +704,9 @@ func (b *BlockChainAPI) GetCode(
 // it returns nil instead of an error since that is according to the API spec,
 // if the error is not of type ErrNotFound it will return the error and the generic
 // empty type.
-func handleError[T any](log zerolog.Logger, err error) (T, error) {
+func handleError[T any](log zerolog.Logger, collector metrics.Collector, err error) (T, error) {
+	collector.ApiErrorOccurred()
```

Comment:
I remember we were discussing that I can log API errors here while you working on handling rate limiter errors here as well. Is that fine for now?